From 99d292b83ebce47ca65d88f890b980c32b16169a Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 9 Nov 2023 15:56:10 +0100
Subject: [PATCH 0563/1695] plat-stm32mp2: enable async notif with GIC PPI 15

Enables OP-TEE async notif (asynchronous notification from OP-TEE
to the non-secure world) using GIC PPI 15 (GIC interrupt line 31).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b1639ad2c992f1c3d8e50adc03d324c2787efd7f plat-stm32mp2: enable async notif with GIC PPI 15]
Change-Id: Ic3707ceb40882ecbcef857bdb2a4b6ca6107f557
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/338807
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 530e1e996..37c3958a9 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -40,6 +40,8 @@ include core/arch/arm/cpu/cortex-armv8-0.mk
 supported-ta-targets ?= ta_arm64
 
 $(call force,CFG_ARM64_core,y)
+$(call force,CFG_CORE_ASYNC_NOTIF,y)
+$(call force,CFG_CORE_ASYNC_NOTIF_GIC_INTID,31)
 $(call force,CFG_DRIVERS_CLK,y)
 $(call force,CFG_DRIVERS_CLK_DT,y)
 $(call force,CFG_DRIVERS_GPIO,y)
-- 
2.47.2

